### ISSUE TITLE
[php7] seg when there is syntax error in model::find condition

### ISSUE THREAD

**Issue by Izopi4a** [phalcon/cphalcon#12038]
test.php

```
<?php
ini_set('display_errors', 1);
error_reporting(E_ALL);

$di = new Phalcon\DI\FactoryDefault();

$di->setShared('db', function () {
    $connection = new Phalcon\Db\Adapter\Pdo\Mysql([
        'adapter'  => 'Mysql',
        'host'     => 'localhost',
        'port'     => 3306,
        'dbname'   => 'test',
        'username' => 'root',
        'password' => "123456"
    ]);

    return $connection;
});

class Robots extends Phalcon\Mvc\Model
{
    public $id;

    public function getSource()
    {
        return 'products';
    }
}

Robots::find([
    'conditions' => 'name == 1' // <- the double ==
]);

```

```

gdb /opt/php-7009/bin/php
GNU gdb (Ubuntu 7.7.1-0ubuntu5~14.04.2) 7.7.1

(gdb) run -e /home/izopi4a/test.php
Starting program: /opt/php-7009/bin/php -e /home/izopi4a/test.php
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007fffe4c96647 in phql_internal_parse_phql () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
(gdb) bt
#0  0x00007fffe4c96647 in phql_internal_parse_phql () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#1  0x00007fffe4c95123 in phql_parse_phql () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#2  0x00007fffe4b3b494 in zim_Phalcon_Mvc_Model_Query_parse () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#3  0x00007fffe47cdd85 in zephir_call_function_opt () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#4  0x00007fffe47c8622 in zephir_call_user_function () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#5  0x00007fffe47c8cdd in zephir_call_class_method_aparams () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#6  0x00007fffe4b4a636 in zim_Phalcon_Mvc_Model_Query_execute () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#7  0x00007fffe47cdd85 in zephir_call_function_opt () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#8  0x00007fffe47c8622 in zephir_call_user_function () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#9  0x00007fffe47c8cdd in zephir_call_class_method_aparams () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#10 0x00007fffe4a9b570 in zim_Phalcon_Mvc_Model_find () from /opt/php-7009/lib/php/extensions/debug-non-zts-20151012/phalcon.so
#11 0x0000000000a80ca4 in ZEND_DO_FCALL_SPEC_HANDLER () at /home/izopi4a/php-7.0.9/Zend/zend_vm_execute.h:842
#12 0x0000000000a7fb26 in execute_ex (ex=0x7fffeec15030) at /home/izopi4a/php-7.0.9/Zend/zend_vm_execute.h:414
#13 0x0000000000a7fc38 in zend_execute (op_array=0x7fffeec80000, return_value=0x0) at /home/izopi4a/php-7.0.9/Zend/zend_vm_execute.h:458
#14 0x0000000000a2406f in zend_execute_scripts (type=8, retval=0x0, file_count=3) at /home/izopi4a/php-7.0.9/Zend/zend.c:1427
#15 0x0000000000990ddb in php_execute_script (primary_file=0x7fffffffd310) at /home/izopi4a/php-7.0.9/main/main.c:2494
#16 0x0000000000ae4ab8 in do_cli (argc=3, argv=0x143ba30) at /home/izopi4a/php-7.0.9/sapi/cli/php_cli.c:974
#17 0x0000000000ae5c77 in main (argc=3, argv=0x143ba30) at /home/izopi4a/php-7.0.9/sapi/cli/php_cli.c:1344
(gdb)


```

i tried for 2 days to fix it myself but failed miserably ...

**Comment by Jurigag** [comment]
In debug phalcon and php it looks like this:

```
0x00007fffe7f6d345 in phql_scanner_error_msg (parser_status=0x7fffefc5d3c0, error_msg=0x7fffffffaa18) at /root/3.0.x-orig/ext/phalcon/mvc/model/query/parser.c:5101
5101            ZVAL_STRING(*error_msg, error);
(gdb) bt
#0  0x00007fffe7f6d345 in phql_scanner_error_msg (parser_status=0x7fffefc5d3c0, error_msg=0x7fffffffaa18) at /root/3.0.x-orig/ext/phalcon/mvc/model/query/parser.c:5101
#1  0x00007fffe7f6e86c in phql_internal_parse_phql (result=0x7fffffffaa08, phql=0x7fffefc7d098 "SELECT name FROM [Robots] WHERE type = `mechanical` LIMIT :APL0:", 
    phql_length=64, error_msg=0x7fffffffaa18) at /root/3.0.x-orig/ext/phalcon/mvc/model/query/parser.c:5546
#2  0x00007fffe7f6d41f in phql_parse_phql (result=0x7fffffffab20, phql=0x7fffffffab10) at /root/3.0.x-orig/ext/phalcon/mvc/model/query/parser.c:5125
#3  0x00007fffe7e13f0a in zim_Phalcon_Mvc_Model_Query_parse (execute_data=0x7fffefc131a0, return_value=0x7fffffffb1e0)
    at /root/3.0.x-orig/ext/phalcon/mvc/model/query.zep.c:3712
#4  0x00007fffe7aa6c26 in zephir_call_function_opt (fci=0x7fffffffadf0, fci_cache=0x7fffffffadc0, info=0x7fffffffaed0, params=0x7fffffffb240)
    at /root/3.0.x-orig/ext/kernel/extended/fcall.c:1154
#5  0x00007fffe7aa13a4 in zephir_call_user_function (object_pp=0x7fffffffb230, obj_ce=0x555555ad4660, type=zephir_fcall_method, function_name=0x0, 
    retval_ptr=0x7fffffffb1e0, cache_entry=0x0, cache_slot=0, param_count=0, params=0x7fffffffb240, info=0x7fffffffaed0) at /root/3.0.x-orig/ext/kernel/fcall.c:455
#6  0x00007fffe7aa1b68 in zephir_call_class_method_aparams (return_value_ptr=0x7fffffffb1e0, ce=0x555555ad4660, type=zephir_fcall_method, object=0x7fffffffb230, 
    method_name=0x7fffe85ddb66 "parse", method_len=5, cache_entry=0x0, cache_slot=0, param_count=0, params=0x7fffffffb240) at /root/3.0.x-orig/ext/kernel/fcall.c:655
#7  0x00007fffe7e230ac in zim_Phalcon_Mvc_Model_Query_execute (execute_data=0x7fffefc13140, return_value=0x7fffffffb760)
    at /root/3.0.x-orig/ext/phalcon/mvc/model/query.zep.c:4990
#8  0x00007fffe7aa6c26 in zephir_call_function_opt (fci=0x7fffffffb4a0, fci_cache=0x7fffffffb470, info=0x7fffffffb580, params=0x7fffffffb7f0)
    at /root/3.0.x-orig/ext/kernel/extended/fcall.c:1154
#9  0x00007fffe7aa13a4 in zephir_call_user_function (object_pp=0x7fffffffb720, obj_ce=0x555555ad4660, type=zephir_fcall_method, function_name=0x0, 
    retval_ptr=0x7fffffffb760, cache_entry=0x0, cache_slot=0, param_count=0, params=0x7fffffffb7f0, info=0x7fffffffb580) at /root/3.0.x-orig/ext/kernel/fcall.c:455
#10 0x00007fffe7aa1b68 in zephir_call_class_method_aparams (return_value_ptr=0x7fffffffb760, ce=0x555555ad4660, type=zephir_fcall_method, object=0x7fffffffb720, 
    method_name=0x7fffe857e1ca "execute", method_len=7, cache_entry=0x0, cache_slot=0, param_count=0, params=0x7fffffffb7f0) at /root/3.0.x-orig/ext/kernel/fcall.c:655
#11 0x00007fffe7d73fe6 in zim_Phalcon_Mvc_Model_find (execute_data=0x7fffefc130d0, return_value=0x7fffefc130b0) at /root/3.0.x-orig/ext/phalcon/mvc/model.zep.c:1308
#12 0x00007ffff3b2cfaa in ZEND_DO_FCALL_SPEC_HANDLER () at /root/.phpbrew/build/7.0.5-debug/Zend/zend_vm_execute.h:842
#13 0x00007ffff3b2be2c in execute_ex (ex=0x7fffefc13030) at /root/.phpbrew/build/7.0.5-debug/Zend/zend_vm_execute.h:414
#14 0x00007ffff3b2bf3e in zend_execute (op_array=0x7fffefc77000, return_value=0x0) at /root/.phpbrew/build/7.0.5-debug/Zend/zend_vm_execute.h:458
#15 0x00007ffff3ad0afe in zend_execute_scripts (type=8, retval=0x0, file_count=3) at /root/.phpbrew/build/7.0.5-debug/Zend/zend.c:1427
#16 0x00007ffff3a3dd69 in php_execute_script (primary_file=0x7fffffffdc70) at /root/.phpbrew/build/7.0.5-debug/main/main.c:2487
#17 0x00007ffff3b9080d in php_handler (r=0x7ffff7ebe0a0) at /root/.phpbrew/build/7.0.5-debug/sapi/apache2handler/sapi_apache2.c:678
#18 0x00005555555aabe0 in ap_run_handler ()
#19 0x00005555555ab129 in ap_invoke_handler ()
#20 0x00005555555c06ca in ap_process_async_request ()
#21 0x00005555555c09a4 in ap_process_request ()
#22 0x00005555555bd442 in ?? ()
#23 0x00005555555b4220 in ap_run_process_connection ()
#24 0x00007ffff4690767 in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#25 0x00007ffff469096c in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#26 0x00007ffff46916b1 in ?? () from /usr/lib/apache2/modules/mod_mpm_prefork.so
#27 0x00005555555918ae in ap_run_mpm ()
#28 0x000055555558b046 in main ()
```

**Comment by andresgutierrez** [comment]
Fixed in the 3.0.x branch
